sled-agent/src/instance.rs

```diff
@@ -237,7 +357,92 @@ struct InstanceInner {
     instance_ticket: InstanceTicket,
 }
 
-impl InstanceInner {
+impl InstanceRunner {
+    async fn run(mut self) -> Result<(), Error> {
+        loop {
+            tokio::select! {
+                biased;
```

Comment:
The priority here is:

1. If we should terminate, do that.
2. Otherwise, keep an eye on any changes self-reported by the VMM.
3. Finally, handle user-requests.